library/src/main/java/com/google/android/exoplayer/text/eia608/Eia608TrackRenderer.java

```diff
@@ -58,16 +67,40 @@
   private final Handler textRendererHandler;
   private final MediaFormatHolder formatHolder;
   private final SampleHolder sampleHolder;
-  private final StringBuilder captionStringBuilder;
+  private final SpannableStringBuilder captionStringBuilder;
   private final TreeSet<ClosedCaptionList> pendingCaptionLists;
 
   private boolean inputStreamEnded;
   private int captionMode;
   private int captionRowCount;
-  private String caption;
-  private String lastRenderedCaption;
+  private SpannableStringBuilder caption;
+  private SpannableStringBuilder lastRenderedCaption;
   private ClosedCaptionCtrl repeatableControl;
 
+  private static final String TAG = "Eia608Captions";
+
+  // Incoming styling commands are ordered, a higher priority command clears any previous lower
```

Comment:
Just for my own education: I assume this is defined in the spec, as opposed to being a decision you made :)?